pym/bob/cmds/archive.py

```diff
@@ -500,6 +500,12 @@ def doArchive(argv, bobRoot):
 """.format(subHelp))
     parser.add_argument("-l", "--local", action='store_true',
                         help="ignore the archive configuration and work on the current directory")
+    parser.add_argument("-a", "--all", action='store_true',
```

Comment:
I think you can use a `add_mutually_exclusive_group` for `-l`, `-a` and `-b`. That way, only one of them can be given and we do not have to define a precedence.